packages/material-ui-lab/src/TreeView/TreeView.js

```diff
@@ -98,31 +109,126 @@ const TreeView = React.forwardRef(function TreeView(props, ref) {
    */
 
   const getNextNode = (id) => {
-    const nodeIndex = visibleNodes.current.indexOf(id);
-    if (nodeIndex !== -1 && nodeIndex + 1 < visibleNodes.current.length) {
-      return visibleNodes.current[nodeIndex + 1];
+    // If expanded get first child
+    if (isExpanded(id) && getChildren(id).length > 0) {
```

Comment:
I've got some changes planned in the future which mean we won't have to check children length here.